src/Section.cpp

```diff
@@ -96,29 +96,26 @@ std::vector<Section> Section::sections(const util::Filter<Section>::type &filter
 
 
 std::vector<Section> Section::findSections(const util::Filter<Section>::type &filter,
-                                           size_t max_depth) const
-{
+                                           size_t max_depth) const {
     std::vector<Section>  results;
     std::list<SectionCont> todo;
-
-    todo.push_back(SectionCont(*this));
-
-    while (todo.size() > 0)
-    {
-        SectionCont current = todo.front();
-        todo.pop_front();
-
-        bool filter_ok = filter(current.entity);
-        if (filter_ok) {
-            results.push_back(current.entity);
+    bool first_iter = true;
+
+    SectionCont current(*this);
+    while (todo.size() > 0 || first_iter) {
+        if(!first_iter) {
```

Comment:
Yeah, I personally think the not-branching-in-while-loop version is more readable.